Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -231,6 +248,62 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
     self.showErrorBannerWithMessage = Signal
       .combineLatest(validateCheckoutExistingCard.errors(), validateCheckoutNewCard.errors())
       .map { _ in Strings.Something_went_wrong_please_try_again() }
+
+    let paymentAuthorizationData: Signal<PostCampaignPaymentAuthorizationData, Never> = self
+      .configureWithDataProperty
+      .signal
+      .skipNil()
+      .map { (data: PostCampaignCheckoutData) -> PostCampaignPaymentAuthorizationData? in
+        guard let firstReward = data.rewards.first else {
+          // There should always be a reward - we create a special "no reward" reward if you make a monetary pledge
+          return nil
+        }
+
+        return PostCampaignPaymentAuthorizationData(
+          project: data.project,
+          hasNoReward: firstReward.isNoReward,
+          subtotal: firstReward.isNoReward ? firstReward.minimum : calculateAllRewardsTotal(
+            addOnRewards: data.rewards,
+            selectedQuantities: data.selectedQuantities
+          ),
+          bonus: data.bonusAmount ?? 0,
+          shipping: data.shipping?.total ?? 0,
+          total: data.total,
```

Comment:
@ifosli This came out of your comment. Here's what I did:
1) Created a new data struct, just for ApplePay for late pledges, which is much more clear about what the different named fields do (subtotal, bonus, shipping, total).
2) Check whether the reward `isNoReward`. Only calculate the subtotal if there is a "real" reward.